File: AdvantageFramework.Core/Database/Entities/ClientContact.cs
﻿using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AdvantageFramework.Core.Database.Entities
{
    [Table("CDP_CONTACT_HDR")]
    public partial class ClientContact
    {
        [Key]
        [Column("CDP_CONTACT_ID")]
        public int CdpContactId { get; set; }
        [Required]
        [Column("CONT_CODE")]
        [StringLength(6)]
        public string ContCode { get; set; }
        [Required]
        [Column("CL_CODE")]
        [StringLength(6)]
        public string ClCode { get; set; }
        [Column("EMAIL_ADDRESS")]
        [StringLength(50)]
        public string EmailAddress { get; set; }
        [Column("CONT_FNAME")]
        [StringLength(30)]
        public string ContFname { get; set; }
        [Column("CONT_LNAME")]
        [StringLength(30)]
        public string ContLname { get; set; }
        [Column("CONT_MI")]
        [StringLength(1)]
        public string ContMi { get; set; }
        [Column("CONT_TITLE")]
        [StringLength(40)]
        public string ContTitle { get; set; }
        [Column("CONT_ADDRESS1")]
        [StringLength(40)]
        public string ContAddress1 { get; set; }
        [Column("CONT_ADDRESS2")]
        [StringLength(40)]
        public string ContAddress2 { get; set; }
        [Column("CONT_CITY")]
        [StringLength(30)]
        public string ContCity { get; set; }
        [Column("CONT_COUNTY")]
        [StringLength(20)]
        public string ContCounty { get; set; }
        [Column("CONT_STATE")]
        [StringLength(10)]
        public string ContState { get; set; }
        [Column("CONT_COUNTRY")]
        [StringLength(20)]
        public string ContCountry { get; set; }
        [Column("CONT_ZIP")]
        [StringLength(10)]
        public string ContZip { get; set; }
        [Column("CONT_TELEPHONE")]
        [StringLength(13)]
        public string ContTelephone { get; set; }
        [Column("CONT_EXTENTION")]
        [StringLength(5)]
        public string ContExtention { get; set; }
        [Column("CONT_FAX")]
        [StringLength(13)]
        public string ContFax { get; set; }
        [Column("CONT_FAX_EXTENTION")]
        [StringLength(5)]
        public string ContFaxExtention { get; set; }
        [Column("SCHEDULE_USER")]
        public short? ScheduleUser { get; set; }
        [Column("DEFAULT_TASK")]
        [StringLength(10)]
        public string DefaultTask { get; set; }
        [Column("CP_USER")]
        public short? CpUser { get; set; }
        [Column("CP_ALERTS")]
        public short? CpAlerts { get; set; }
        [Column("EMAIL_RCPT")]
        public short? EmailRcpt { get; set; }
        [Column("TASK_PRIMARY")]
        public short? TaskPrimary { get; set; }
        [Column("INACTIVE_FLAG")]
        public short? InactiveFlag { get; set; }
        [Column("CONT_LF")]
        [StringLength(62)]
        public string ContLf { get; set; }
        [Column("CONT_FML")]
        [StringLength(64)]
        public string ContFml { get; set; }
        [Column("CELL_PHONE")]
        [StringLength(13)]
        public string CellPhone { get; set; }
        [Column("CONT_COMMENT", TypeName = "text")]
        public string ContComment { get; set; }
        [Column("CONTACT_TYPE_ID")]
        public int? ContactTypeId { get; set; }
    }
}

File: AdvantageFramework/Security/Database/Entities/ClientContact.vb
Namespace Security.Database.Entities

    <Table("CDP_CONTACT_HDR")>
    Public Class ClientContact
        Inherits BaseClasses.Entity

#Region " Constants "



#End Region

#Region " Enum "

        Public Enum Properties
            ContactID
            ContactCode
            ClientCode
            EmailAddress
            FirstName
            LastName
            MiddleInitial
            Title
            Address
            Address2
            City
            County
            State
            Country
            Zip
            PhoneNumber
            PhoneNumberExtention
            FaxNumber
            FaxNumberExtention
            ScheduleUser
            DefaultTaskCode
            RecievesAlerts
            IsEmailRecipient
            IsPrimaryTask
            IsInactive
            CellPhoneNumber
            Comments
            IsClientPortalUser
            ContactTypeID
            ClientContactDetails
            Client
            ClientPortalUsers
            ClientPortalUserWorkspaces
        End Enum

#End Region

#Region " Variables "



#End Region

#Region " Properties "

        <Key>
        <System.ComponentModel.DataAnnotations.Schema.DatabaseGeneratedAttribute(ComponentModel.DataAnnotations.Schema.DatabaseGeneratedOption.Identity)>
        <Required>
        <Column("CDP_CONTACT_ID")>
        <AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", ShowColumnInGrid:=False)>
        Public Property ContactID() As Integer
        <Required>
        <MaxLength(6)>
		<Column("CONT_CODE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=True, DisplayFormat:="", PropertyType:=AdvantageFramework.BaseClasses.PropertyTypes.Code)>
		Public Property ContactCode() As String
		<Required>
		<MaxLength(6)>
		<Column("CL_CODE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=True, DisplayFormat:="", ShowColumnInGrid:=True, PropertyType:=AdvantageFramework.BaseClasses.PropertyTypes.ClientCode)>
		Public Property ClientCode() As String
		<MaxLength(50)>
		<Column("EMAIL_ADDRESS", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", PropertyType:=AdvantageFramework.BaseClasses.PropertyTypes.Email)>
		Public Property EmailAddress() As String
		<MaxLength(30)>
		<Column("CONT_FNAME", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property FirstName() As String
		<MaxLength(30)>
		<Column("CONT_LNAME", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property LastName() As String
		<MaxLength(1)>
		<Column("CONT_MI", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property MiddleInitial() As String
		<MaxLength(40)>
		<Column("CONT_TITLE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Title() As String
		<MaxLength(40)>
		<Column("CONT_ADDRESS1", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Address() As String
		<MaxLength(40)>
		<Column("CONT_ADDRESS2", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Address2() As String
		<MaxLength(30)>
		<Column("CONT_CITY", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property City() As String
		<MaxLength(20)>
		<Column("CONT_COUNTY", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property County() As String
		<MaxLength(10)>
		<Column("CONT_STATE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property State() As String
		<MaxLength(20)>
		<Column("CONT_COUNTRY", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Country() As String
		<MaxLength(10)>
		<Column("CONT_ZIP", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Zip() As String
		<MaxLength(13)>
		<Column("CONT_TELEPHONE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property PhoneNumber() As String
		<MaxLength(5)>
		<Column("CONT_EXTENTION", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property PhoneNumberExtention() As String
		<MaxLength(13)>
		<Column("CONT_FAX", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property FaxNumber() As String
		<MaxLength(5)>
		<Column("CONT_FAX_EXTENTION", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property FaxNumberExtention() As String
		<Column("SCHEDULE_USER")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", ShowColumnInGrid:=True, DefaultColumnType:=AdvantageFramework.BaseClasses.DefaultColumnTypes.StandardCheckBox)>
		Public Property ScheduleUser() As Nullable(Of Short)
		<MaxLength(10)>
		<Column("DEFAULT_TASK", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", ShowColumnInGrid:=False)>
		Public Property DefaultTaskCode() As String
		<Column("CP_ALERTS")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", ShowColumnInGrid:=True, DefaultColumnType:=AdvantageFramework.BaseClasses.DefaultColumnTypes.StandardCheckBox)>
		Public Property RecievesAlerts() As Nullable(Of Short)
		<Column("EMAIL_RCPT")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", ShowColumnInGrid:=True, DefaultColumnType:=AdvantageFramework.BaseClasses.DefaultColumnTypes.StandardCheckBox)>
		Public Property IsEmailRecipient() As Nullable(Of Short)
		<Column("TASK_PRIMARY")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", ShowColumnInGrid:=False)>
		Public Property IsPrimaryTask() As Nullable(Of Short)
		<Column("INACTIVE_FLAG")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", ShowColumnInGrid:=True, DefaultColumnType:=AdvantageFramework.BaseClasses.DefaultColumnTypes.StandardCheckBox)>
		Public Property IsInactive() As Nullable(Of Short)
		<MaxLength(13)>
		<Column("CELL_PHONE", TypeName:="varchar")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property CellPhoneNumber() As String
		<Column("CONT_COMMENT")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="")>
		Public Property Comments() As String
		<Column("CP_USER")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", ShowColumnInGrid:=True, DefaultColumnType:=AdvantageFramework.BaseClasses.DefaultColumnTypes.StandardCheckBox)>
		Public Property IsClientPortalUser() As Nullable(Of Short)
		<Column("CONTACT_TYPE_ID")>
		<AdvantageFramework.BaseClasses.Attributes.Entity(IsRequired:=False, DisplayFormat:="", PropertyType:=AdvantageFramework.BaseClasses.PropertyTypes.ContactTypeID)>
		Public Property ContactTypeID() As Nullable(Of Integer)
		<Column("CONT_LF", TypeName:="varchar")>
		Public Property FullNameLF() As String
		<Column("CONT_FML", TypeName:="varchar")>
		Public Property FullNameFML() As String

        Public Overridable Property ClientContactDetails() As ICollection(Of Database.Entities.ClientContactDetail)
        Public Overridable Property Client As Database.Entities.Client
        Public Overridable Property ClientPortalUsers() As ICollection(Of Database.Entities.ClientPortalUser)
        Public Overridable Property ClientPortalUserWorkspaces() As ICollection(Of Database.Entities.ClientPortalUserWorkspace)

#End Region

#Region " Methods "

        Public Overrides Function ToString() As String

            ToString = Me.ContactID

        End Function

#End Region

    End Class

End Namespace
